arango/collection.py

```diff
@@ -1934,7 +1934,8 @@ def import_bulk(
         overwrite: Optional[bool] = None,
         on_duplicate: Optional[str] = None,
         sync: Optional[bool] = None,
-    ) -> Result[Json]:
+        batch_size: Optional[int] = None,
+    ) -> Union[Result[Json], List[Result[Json]]]:
```

Comment:
I'm inclined to add a new method "import_bulk _by_batches" just to avoid the `Union` type. I'm not a fan of a function returning different types if it can be avoided.

Reply:
As an alternative to this, would it be worth looking into converting `List[Result[Json]]` into a `Result[Json]` (i.e combined all "batch" `import_bulk` results into one)

Reply:
Actually on second thought I think the way you have it is the best. Let's keep it but clearly document in the docstring that the output type will change when you set `batch_size`.

Reply:
> Actually on second thought I think the way you have it is the best. Let's keep it but clearly document in the docstring that the output type will change when you set batch_size.

Addressed in 56d129f. Feel free to resolve if this change is suitable